addon/components/frost-date-picker.js

```diff
@@ -68,7 +94,8 @@ export default Component.extend(EventsProxyMixin, {
     run.scheduleOnce('sync', this, function () {
       const pikadayOptions = {
         field: this.$('input')[0],
-        onSelect: run.bind(this, this._onSelect)
+        onSelect: run.bind(this, this._onSelect),
```

Comment:
I'm usually paranoid about these things. We've had issues in the past with memory leaks due to innocent pieces of code like this. You're probably safe to leave it as is since pickaday shouldn't be keeping the reference when the instance is destroyed. Keyword being "shouldn't".